client/rest-high-level/src/main/java/org/elasticsearch/client/eql/EqlSearchResponse.java

```diff
@@ -165,17 +170,27 @@ public int hashCode() {
                     return BytesReference.bytes(builder);
                 }
             }, SOURCE);
+            PARSER.declareObject(optionalConstructorArg(), (p, c) -> {
+                Map<String, DocumentField> fields = new HashMap<>();
+                while (p.nextToken() != XContentParser.Token.END_OBJECT) {
+                    DocumentField field = DocumentField.fromXContent(p);
+                    fields.put(field.getName(), field);
+                }
+                return fields;
+            }, FIELDS);
         }
 
         private final String index;
         private final String id;
         private final BytesReference source;
         private Map<String, Object> sourceAsMap;
+        private final Map<String, DocumentField> fetchFields;
 
-        public Event(String index, String id, BytesReference source) {
+        public Event(String index, String id, BytesReference source, Map<String, DocumentField> fetchFields) {
```

Comment:
Since this is a public constructor, changing the signature breaks backwards compatibility for existing clients; the existing constructor needs to be kept in place while delegating to the new one.

Reply:
Agreed